src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -523,8 +523,13 @@ size_t shim::xclRead(xclAddressSpace space, uint64_t offset, void *hostBuf, size
 unsigned int shim::xclAllocBO(size_t size, int unused, unsigned flags)
 {
     drm_xocl_create_bo info = {size, mNullBO, flags};
+    unsigned int bo = mNullBO;
     int result = mDev->ioctl(mUserHandle, DRM_IOCTL_XOCL_CREATE_BO, &info);
-    return result ? mNullBO : info.handle;
+    if (result)
+        errno = result;
+    else
+        bo = info.handle;
+    return bo;
```

Comment:
This does not look right.   ioctl should return -1 on error and errno should be set.   You are overwriting it with -1.

Reply:
The alloc bo ioctls return the error code from the driver. So far we have been throwing away that information which does not help the client since it only sees a null handle when the API fails.

Reply:
Ok, that is wrong then; it should set errno according to the ioctl man page.